creator-node/src/services/sync/secondarySyncFromPrimary.js

```diff
@@ -108,6 +134,16 @@ const handleSyncFromPrimary = async ({
       localMaxClockVal = cnodeUser ? cnodeUser.clock : -1
     }
 
+    // Ensure this node is one of the user's secondaries (except when wiping a node with orphaned data)
+    if (
+      thisContentNodeEndpoint !== userReplicaSet[1] &&
+      thisContentNodeEndpoint !== userReplicaSet[2]
+    ) {
+      throw new Error(
+        `This node is not one of the user's secondaries. This node: ${thisContentNodeEndpoint} Secondaries: [${userReplicaSet?.[1]},${userReplicaSet?.[2]}]`
+      )
+    }
+
```

Comment:
some important new constraints added here - can you document in PR desc
cc @dmanjunath for visibility

Reply:
thanks for flagging @SidSethi. i actually don't think we need this condition, my take on how to do conditions in to combine the secondary check with forceResync and forceWipe

eg forceWipe should only be called on non-secondaries. forceResync can be called on any node but we can lock it down to secondaries only. but imo no need to add this error condition here

Reply:
> my take on how to do conditions in to combine the secondary check with forceResync and forceWipe
> 
> eg forceWipe should only be called on non-secondaries. forceResync can be called on any node but we can lock it down to secondaries only

this currently makes forceWipe and forceResync only work on non-primaries, but I can change it to make:
* forceWipe only work on non-RS nodes
* forceResync only work on secondaries

are you saying you don't want to remove checks to make:
* forceResync and forceWipe never work on a primary? (I guess this becomes redundant with the above checks so makes sense to remove)
* all secondarySyncFromPrimary calls only work when this node is secondary and node synced from is primary? (this seems like something that's good to have)

Reply:
@SidSethi @dmanjunath I updated the PR description after syncing offline. here's a recap with code pointers -- please lmk if you want any changes here:

- issueSyncRequest job processor for sync types of `MergePrimaryAndSecondary` and `MergePrimaryThenWipeSecondary`: enforces [the node processing the job being primary](https://github.com/AudiusProject/audius-protocol/blob/theo-orphaned-data-job/creator-node/src/services/stateMachineManager/stateReconciliation/issueSyncRequest.jobProcessor.ts#L550-L554) and the [node endpoint issued to being secondary](https://github.com/AudiusProject/audius-protocol/blob/theo-orphaned-data-job/creator-node/src/services/stateMachineManager/stateReconciliation/issueSyncRequest.jobProcessor.ts#L556-L563) (for MergePrimaryAndSecondary) or [outside the replica set](https://github.com/AudiusProject/audius-protocol/blob/theo-orphaned-data-job/creator-node/src/services/stateMachineManager/stateReconciliation/issueSyncRequest.jobProcessor.ts#L565-L575) (for MergePrimaryThenWipeSecondary)
- secondarySyncFromPrimary:
    - enforces [syncing *from* primary](https://github.com/AudiusProject/audius-protocol/blob/theo-orphaned-data-job/creator-node/src/services/sync/secondarySyncFromPrimary.js#L53-L65) (i.e., the node being synced from -- endpoint that it sends `/export` request to -- being primary)
    - enforces [syncing *to* non-primary if it's being wiped](https://github.com/AudiusProject/audius-protocol/blob/theo-orphaned-data-job/creator-node/src/services/sync/secondarySyncFromPrimary.js#L90-L97) (i.e. the current node executing the operation is not primary if forceWipe=true or forceResync=true)
    - enforces [syncing *to* secondary except when wiping an orphaned node](https://github.com/AudiusProject/audius-protocol/blob/theo-orphaned-data-job/creator-node/src/services/sync/secondarySyncFromPrimary.js#L135-L144) (i.e., the current node executing the operation is secondary if forceWipe=false)#3696

Reply:
amazing - really helpful writeup as well